src/policy/concrete.rs

```diff
@@ -681,13 +681,21 @@ impl<Pk: MiniscriptKey> Policy<Pk> {
     /// and [`Policy::Threshold`] disjunctions for the `TapTree`.
     #[cfg(feature = "compiler")]
     fn num_tap_leaves(&self) -> usize {
-        match self {
-            Policy::Or(subs) => subs.iter().map(|(_prob, pol)| pol.num_tap_leaves()).sum(),
-            Policy::Threshold(k, subs) if *k == 1 => {
-                subs.iter().map(|pol| pol.num_tap_leaves()).sum()
-            }
-            _ => 1,
+        use Policy::*;
+
+        let mut nums = vec![];
+        for data in Arc::new(self).post_order_iter() {
+            let num_for_child_n = |n| nums[data.child_indices[n]];
+
+            let num = match data.node {
+                Or(subs) => (0..subs.len()).map(num_for_child_n).sum(),
+                Threshold(k, subs) if *k == 1 => (0..subs.len()).map(num_for_child_n).sum(),
+                _ => 1,
+            };
+            nums.push(num);
```

Comment:
In 7e73ffeb14a5e95a019c6abcb72d6976391e2969:

We don't need to maintain a vector of numbers here. We can just do a running sum and avoid some allocations and unwraps.

Reply:
I'm not sure we can, don't we need to sum the results of nested `Or`s? I don't think we can do that without recursion or keeping the nums vec. Am I correct in thinking a tree like this should return 4
```
        or
      /     \ 
   or       or
/ \   /\  /\  /\
k  k k k  k  k  k
```

Reply:
Yep, you're right, I looked at the algorithm too quickly.

Reply:
No sweat, it made me double check it which was not a bad thing :)